src/Microsoft.Win32.Primitives/src/System/ComponentModel/Win32Exception.cs

```diff
@@ -7,80 +7,57 @@
-        ///    specified detailed description.</para>
-        /// </devdoc>
-        public Win32Exception(int error, string message)
-        : base(message)
+        /// <summary>
+        /// Initializes a new instance of the <see cref='System.ComponentModel.Win32Exception'/> class with the specified error and the 
+        /// specified detailed description.
+        /// </summary>
+        public Win32Exception(int error, string message) : base(message)
         {
-            nativeErrorCode = error;
+            NativeErrorCode = error;
         }
 
-        /// <devdoc>
-        ///     Initializes a new instance of the Exception class with a specified error message.
-        ///     FxCop CA1032: Multiple constructors are required to correctly implement a custom exception.
-        /// </devdoc>
+        /// <summary>
+        /// Initializes a new instance of the Exception class with a specified error message.
+        /// </summary>
         public Win32Exception(string message) : this(Marshal.GetLastWin32Error(), message)
         {
         }
 
-        /// <devdoc>
-        ///     Initializes a new instance of the Exception class with a specified error message and a 
-        ///     reference to the inner exception that is the cause of this exception.
-        ///     FxCop CA1032: Multiple constructors are required to correctly implement a custom exception.
-        /// </devdoc>
+        /// <summary>
+        /// Initializes a new instance of the Exception class with a specified error message and a 
+        /// reference to the inner exception that is the cause of this exception.
+        /// </summary>
         public Win32Exception(string message, Exception innerException) : base(message, innerException)
         {
-            nativeErrorCode = Marshal.GetLastWin32Error();
+            NativeErrorCode = Marshal.GetLastWin32Error();
         }
 
-        protected Win32Exception(SerializationInfo info, StreamingContext context) : base(info, context)
-        {
-            throw new PlatformNotSupportedException();
-        }
+        protected Win32Exception(SerializationInfo info, StreamingContext context) : base(info, context) { }
 
-        public override void GetObjectData(SerializationInfo info, StreamingContext context)
-        {
-            base.GetObjectData(info, context);
-        }
-
-        /// <devdoc>
-        ///    <para>Represents the Win32 error code associated with this exception. This 
-        ///       field is read-only.</para>
-        /// </devdoc>
-        public int NativeErrorCode
-        {
-            get
-            {
-                return nativeErrorCode;
-            }
-        }
+        /// <summary>
+        /// Represents the Win32 error code associated with this exception. This field is read-only.
+        /// </summary>
+        public int NativeErrorCode { get; }
```

Comment:
Before, I removed the dead serialization constructor, and then realised that I could clean up Win32Exception at the same time as adding some tests. @AtsushiKan suggested some more cleanup items while I was at it